services/app-web/src/pages/StateSubmission/RateDetails/V2/RateDetailsV2.tsx

```diff
@@ -199,70 +147,75 @@ const RateDetailsV2 = ({
         },
         skip: !displayAsStandaloneRate,
     })
-    const ratesFromContract =
-        fetchContractData?.fetchContract.contract.draftRates // TODO WHEN WE IMPLEMENT UDPATE API, THIS SHOULD ALSO LOAD FROM ANY LINKED RATES
-    const initialRequestLoading = fetchContractLoading || fetchRateLoading
-    const initialRequestError = fetchContractError || fetchRateError
-    const previousDocuments: string[] = []
-
-    React.useEffect(() => {
+    useEffect(() => {
         if (focusNewRate) {
             newRateNameRef?.current?.focus()
             setFocusNewRate(false)
             newRateNameRef.current = null
         }
     }, [focusNewRate])
 
-    /*
-    Set up initial rate form values for Formik
-        if contract rates exist, use those (relevant for multi rate forms on contract package submission form)
-        if standalone rates exist, use those (for a standalone rate edits)
-        otherwise, generate a new  list of empty rate form values
-    */
-    const rates: Rate[] = React.useMemo(
+    const [updateDraftContractRates, { error: updateContractError }] =
+        useUpdateDraftContractRatesMutation()
+    const [submitRate, { error: submitRateError }] = useSubmitRateMutation()
+
+    // Set up data for form. Either based on contract API (for multi rate) or rates API (for edit and submit of standalone rate)
+    const ratesFromContract =
+        fetchContractData?.fetchContract.contract.draftRates // TODO WHEN WE IMPLEMENT UPDATE API, THIS SHOULD ALSO LOAD FROM ANY LINKED RATES
+    const initialRequestLoading = fetchContractLoading || fetchRateLoading
```

Comment:
Ok, now I think I see where you are skipping requests. I'm a little surprised by this pattern, seems like a lot of overhead is required in order to use this component for both stand alone rate edit and inside of the RateDetails array. No need to change this now but I'd like to look at that after summit again.